syntax = "proto3"; // we use proto version 3
package consensusInterface;   // similar to namespace, avoid conflicts

/** Paxos consensus **/
service ConsensusService { 
  rpc propose(Request) returns (Response) {};
  rpc accept(Request) returns (Response) {};
  rpc success(Request) returns (Response) {};
  rpc heartbeat(Request) returns (Response) {};
}

message Request {
    string s = 1;
}

message Response {
    string m = 2;
}


